Google.Api.Generator.Rest/Models/MethodModel.cs

```diff
@@ -15,25 +15,84 @@
 using Google.Api.Generator.Utils;
 using Google.Api.Generator.Utils.Roslyn;
 using Google.Apis.Discovery.v1.Data;
+using Google.Apis.Services;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.Collections.Generic;
+using System.Linq;
 using static Google.Api.Generator.Utils.Roslyn.RoslynBuilder;
 
 namespace Google.Api.Generator.Rest.Models
 {
     public class MethodModel
     {
+        private RestMethod _restMethod;
+
         public PackageModel Package { get; }
+        /// <summary>
+        /// Resource that owns this method, if any.
+        /// </summary>
+        public ResourceModel Resource { get; }
+        
+        public IReadOnlyList<ParameterModel> Parameters { get; }
+        public Typ ParentTyp { get; }
+        public Typ RequestTyp { get; }
         public string Name { get; }
 
-        public MethodModel(PackageModel package, string name, RestMethod restMethod)
+        public MethodModel(PackageModel package, ResourceModel resource, string name, RestMethod restMethod)
         {
             Package = package;
+            Resource = resource;
```

Comment:
I think you are not using this anywhere else, I'd say, remove it, if it's not relevant for MethodModel then it shouldn't be relevant outside, or if it becomes relevant then maybe MethodModel should encapsulate that.

Reply:
I'm generally being fairly lax about what to define as properties. It makes things a lot quicker for development. I suggest that we don't worry about it for the moment, but that we go back over all the models later to see what we actually need to expose. It's definitely inconsistent right now.